runtime/src/accounts_index.rs

```diff
@@ -707,10 +712,12 @@ impl<'a, T: 'static + Clone> Iterator for AccountsIndexIterator<'a, T> {
         let (start_bin, bin_range) = self.bin_start_and_range();
         let mut chunk: Vec<(Pubkey, AccountMapEntry<T>)> = Vec::with_capacity(ITER_BATCH_SIZE);
         'outer: for i in self.account_maps.iter().skip(start_bin).take(bin_range) {
-            for (pubkey, account_map_entry) in
-                Self::range(&i.read().unwrap(), (self.start_bound, self.end_bound))
-            {
-                if chunk.len() >= ITER_BATCH_SIZE {
+            for (pubkey, account_map_entry) in Self::range(
+                &i.read().unwrap(),
+                (self.start_bound, self.end_bound),
+                self.collect_all_unsorted,
+            ) {
+                if chunk.len() >= ITER_BATCH_SIZE && !self.collect_all_unsorted {
```

Comment:
@ryoqun, and for rent collection we don't need to limit the # of results we return at once.

Reply:
yes too. (ideally, we would like to parallelize the eager rent collection.. or change the whole rent collection mechanism altogether with @aeyakovenko 's account.deletion_slot based rent collection approch...)

Reply:
@carllin @sakridge - this is a commit that gets rid of the sorting for rent collection to understand that more deeply. Initial bench-tps/perf-test numbers indicate this is better than current master. I don't have accurate numbers yet. There is a regression in master that behzad is looking at.